universe/Conditions.cpp

```diff
@@ -640,7 +633,21 @@ SortedNumberOf::SortedNumberOf(std::unique_ptr<ValueRef::ValueRef<int>>&& number
     m_sort_key(std::move(sort_key_ref)),
     m_sorting_method(sorting_method),
     m_condition(std::move(condition))
-{}
+{
+    m_root_candidate_invariant =
+        (!m_number || m_number->SourceInvariant()) &&
```

Comment:
Should probably be calling `RootCandidateInvariant` not `SourceInvariant` here. (Not original in your changes, but should be fixed)